### PR TITLE
Bump rayon from 1.3.0 to 1.5.0

### PR DESCRIPTION
**Pull request by kylefleming**
Bumps [rayon](https://github.com/rayon-rs/rayon) from 1.3.0 to 1.5.0.
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a href="https://github.com/rayon-rs/rayon/blob/master/RELEASES.md">rayon's changelog</a>.</em></p>
<blockquote>
<h1>Release rayon 1.5.0 / rayon-core 1.9.0 (2020-10-21)</h1>
<ul>
<li>Update crossbeam dependencies.</li>
<li>The minimum supported <code>rustc</code> is now 1.36.</li>
</ul>
<h2>Contributors</h2>
<p>Thanks to all of the contributors for this release!</p>
<ul>
<li><a href="https://github.com/cuviper">@cuviper</a></li>
<li><a href="https://github.com/mbrubeck">@mbrubeck</a></li>
<li><a href="https://github.com/mrksu">@mrksu</a></li>
</ul>
<h1>Release rayon 1.4.1 (2020-09-29)</h1>
<ul>
<li>The new <code>flat_map_iter</code> and <code>flatten_iter</code> methods can be used to flatten
sequential iterators, which may perform better in cases that don't need the
nested parallelism of <code>flat_map</code> and <code>flatten</code>.</li>
<li>The new <code>par_drain</code> method is a parallel version of the standard <code>drain</code> for
collections, removing items while keeping the original capacity. Collections
that implement this through <code>ParallelDrainRange</code> support draining items from
arbitrary index ranges, while <code>ParallelDrainFull</code> always drains everything.</li>
<li>The new <code>positions</code> method finds all items that match the given predicate and
returns their indices in a new iterator.</li>
</ul>
<h1>Release rayon-core 1.8.1 (2020-09-17)</h1>
<ul>
<li>Fixed an overflow panic on high-contention workloads, for a counter that was
meant to simply wrap. This panic only occurred with debug assertions enabled,
and was much more likely on 32-bit targets.</li>
</ul>
<h1>Release rayon 1.4.0 / rayon-core 1.8.0 (2020-08-24)</h1>
<ul>
<li>Implemented a new thread scheduler, <a href="https://github-redirect.dependabot.com/rayon-rs/rfcs/pull/5">RFC 5</a>, which uses targeted wakeups for
new work and for notifications of completed stolen work, reducing wasteful
CPU usage in idle threads.</li>
<li>Implemented <code>IntoParallelIterator for Range&lt;char&gt;</code> and <code>RangeInclusive&lt;char&gt;</code>
with the same iteration semantics as Rust 1.45.</li>
<li>Relaxed the lifetime requirements of the initial <code>scope</code> closure.</li>
</ul>
<h2>Contributors</h2>
<p>Thanks to all of the contributors for this release!</p>
<ul>
<li><a href="https://github.com/CAD97">@CAD97</a></li>
<li><a href="https://github.com/cuviper">@cuviper</a></li>
<li><a href="https://github.com/kmaork">@kmaork</a></li>
<li><a href="https://github.com/nikomatsakis">@nikomatsakis</a></li>
</ul>

</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://github.com/rayon-rs/rayon/commit/b8b97a17bc4cbef89807444566eee7cdc523b7d1"><code>b8b97a1</code></a> Merge <a href="https://github-redirect.dependabot.com/rayon-rs/rayon/issues/661">#661</a></li>
<li><a href="https://github.com/rayon-rs/rayon/commit/3462f0be6628c97c5c00ffa3d7b7747246754359"><code>3462f0b</code></a> add me since I fixed a typo :)</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/9feaac61d794a7f4b8466163e02a21bc03f2d4be"><code>9feaac6</code></a> Update RELEASES.md</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/06b8c618a9964d68ca59f6c59e5110d8a71a8a51"><code>06b8c61</code></a> Release rayon 1.1.0 / rayon-core 1.5.0</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/c9f4c8ce678bf704d14a5db97fafc540f0085d59"><code>c9f4c8c</code></a> Add dates to past releases</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/49fb38a5b797fe1e1234cb787c16cb76b18621d9"><code>49fb38a</code></a> Merge <a href="https://github-redirect.dependabot.com/rayon-rs/rayon/issues/636">#636</a></li>
<li><a href="https://github.com/rayon-rs/rayon/commit/249ad3f0efefac6fbd5918b56f9aa32523ae68a1"><code>249ad3f</code></a> Clear the current thread when WorkerThread drops</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/7c4dcd9c86d7a49eb44ddc9d3ec44addc5fb2712"><code>7c4dcd9</code></a> Add links and examples to build_scoped and spawn_handler</li>
<li><a href="https://github.com/rayon-rs/rayon/commit/12afe12f0e720e5acd677f0be2459e622701770e"><code>12afe12</code></a> Merge <a href="https://github-redirect.dependabot.com/rayon-rs/rayon/issues/660">#660</a></li>
<li><a href="https://github.com/rayon-rs/rayon/commit/da6dbeaa3d5564253513d8ccc5b7485b5d8aebec"><code>da6dbea</code></a> docs: remove &quot;double wording&quot;</li>
<li>Additional commits viewable in <a href="https://github.com/rayon-rs/rayon/compare/rayon-core-v1.3.0...rayon-core-v1.5.0">compare view</a></li>
</ul>
</details>
<br />

### CODE REVIEW

**Review by sugargoat — APPROVED** [review]

**Review by jcape — CHANGES_REQUESTED** [review]
This should absolutely be tested in a live environment and slammed before we merge it.

It includes a rewritten thread scheduler, which appears to be used inside byzantine ledger for transaction management stuff---it's not too hard to envision a scenario where a bug in this new scheduler leads to node lockup.

**Comment by kylefleming** [comment]
Deploying as [build/84](https://deploybot.canadacentral.mobilecoin.com/job/mobilecoin-internal/job/deploy%252Fbuild/84/)

**Comment by kylefleming** [comment]
Deployed successfully as [build/85](https://deploybot.canadacentral.mobilecoin.com/job/mobilecoin-internal/job/deploy%252Fbuild/85/)

**Review by jcape — APPROVED** [review]
Approved (pending merge conflict resolution).

**Comment by kylefleming** [comment]
Rebased on master.

**Comment by kylefleming** [comment]
Closing, as more testing is needed, including running slam. Another PR should be created whenever more attention can be given to this upgrade.

**Pull request closed without merging**
